packages/drafts-realm/cycling-mileage-log.gts

```diff
@@ -0,0 +1,148 @@
+import {
+  contains,
+  containsMany,
+  CardDef,
+  FieldDef,
+  StringField,
+  field,
+} from 'https://cardstack.com/base/card-api';
+import NumberField from 'https://cardstack.com/base/number';
+import { Component } from 'https://cardstack.com/base/card-api';
+import Date from 'https://cardstack.com/base/date';
+// @ts-ignore
+import {
+  Chart,
+  registerables,
+  // @ts-ignore
+} from 'https://cdn.jsdelivr.net/npm/chart.js@4.4.2/+esm';
+import Modifier from 'ember-modifier';
+
+type ProgressChartModifierSignature = {
+  Args: {
+    Named: {
+      target?: number;
+      total?: number;
+    };
+  };
+  Element: HTMLElement;
+};
+
+class ProgressChartModifier extends Modifier<ProgressChartModifierSignature> {
+  modify(
+    element: HTMLElement,
+    _positional: [],
+    { target, total }: ProgressChartModifierSignature['Args']['Named'],
+  ) {
+    total = total ?? 0;
+    target = target ?? 0;
+    Chart.register(...registerables);
+    new Chart(element, {
+      type: 'doughnut',
+      data: {
+        labels: ['progress (KM)', 'remaining (KM)'],
+        datasets: [
+          {
+            label: ['Cycling Mileage Log'],
+            data: [total, target - total],
+            backgroundColor: ['rgb(255, 99, 132)', 'rgb(54, 162, 235)'],
```

Comment:
tip for better design - make the color that indicates the remaining mileage more neutral  (white or gray) so it doesn't interfere with the progress color

<img width="750" alt="image" src="https://github.com/cardstack/boxel/assets/273660/9a77911d-d231-4e3f-a4f6-5938836a237d">